### PR TITLE
feat: logging interface

### PR DESCRIPTION
**Pull request by sudeshrshetty**
-provided logging interface
-log levels supported - CRITICAL, ERROR, WARNING, INFO, DEBUG, where
INFO is default logging level
-logging provider implementation can be used to provide custom logging
provider
-Story: #21

Signed-off-by: sudesh.shetty <sudesh.shetty@securekey.com>

### CODE REVIEW

File: pkg/common/logging/api/loggerapi.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/*
```
Comment (troyronda) [inline pkg/common/logging/api/loggerapi.go]: is the api subdirectory needed? is there a more descriptive name for the package?

File: pkg/common/logging/api/loggerapi.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	ERROR
+	WARNING
+	INFO //default logging level
+	DEBUG
+)
+
+//Logger - Standard logger interface
+type Logger interface {
+
+	//Fatalf is critical fatal logging, should possibly followed by system shutdown
```
Comment (rolsonquadras) [inline pkg/common/logging/api/loggerapi.go]: "should possibly followed by system shutdown" is bit confusing. May be we can use standard Go library doc message (to be consistent with the standard lib).

"Fatalf is equivalent to Printf() followed by a call to os.Exit(1)" - https://godoc.org/log#Fatalf

File: pkg/common/logging/api/loggerapi.go (hunk 1 above)
Comment (sudeshrshetty) [reply pkg/common/logging/api/loggerapi.go]: renamed 'common/logging' to 'common/log', and moved logger interfaces right inside 'common/log'.

File: pkg/common/logging/api/loggerapi.go (hunk 2 above)
Comment (sudeshrshetty) [reply pkg/common/logging/api/loggerapi.go]: Fixed

**Pull request merged**